docker/builder/builder.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+
+# Start container with required tools to build packages
+# Requires Docker
+# Script usage: bash ./builder.sh
+
+set -e
+
+# ====
+# Checks that the script is run from the intended location
+# ====
+function check_project_root_folder() {
+    current=$(basename "$(pwd)")
+
+    if [[ "$1" != "./builder.sh" && "$1" != "builder.sh" ]]; then
+        echo "Run the script from its location"
+        usage
+        exit 1
+    fi
+    # Change working directory to the root of the repository
+    cd ../..
+}
+
+# ====
+# Parse arguments
+# ====
+function parse_args() {
+
+    while getopts ":p:r:R:s:d:a:Dh" arg; do
+        case $arg in
+        h)
+            usage
+            exit 1
+            ;;
+        p)
+            INDEXER_PLUGINS_BRANCH=$OPTARG
+            ;;
+        r)
+            INDEXER_REPORTING_BRANCH=$OPTARG
+            ;;
+        R)
+            REVISION=$OPTARG
+            ;;
+        s)
+            IS_STAGE=$OPTARG
+            ;;
+        d)
+            DISTRIBUTION=$OPTARG
+            ;;
+        a)
+            ARCHITECTURE=$OPTARG
+            ;;
+        D)
+            DESTROY=$OPTARG
+            ;;
+        :)
+            echo "Error: -${OPTARG} requires an argument"
+            usage
+            exit 1
+            ;;
+        ?)
+            echo "Invalid option: -${arg}"
+            exit 1
+            ;;
+        esac
+    done
+
+    ## Set defaults:
+    [ -z "$INDEXER_PLUGINS_BRANCH" ] && INDEXER_PLUGINS_BRANCH="master"
+    [ -z "$INDEXER_REPORTING_BRANCH" ] && INDEXER_REPORTING_BRANCH="master"
+    [ -z "$REVISION" ] && REVISION="0"
+    [ -z "$IS_STAGE" ] && IS_STAGE="false"
+    [ -z "$DISTRIBUTION" ] && DISTRIBUTION="rpm"
+    [ -z "$ARCHITECTURE" ] && ARCHITECTURE="x64"
+    [ -z "$DESTROY" ] && DESTROY=false
+}
+
+
+# ====
+# Displays usage
+# ====
+function usage() {
+  echo "Usage: $0 [args]"
+  echo ""
+  echo "Arguments:"
+  echo -e "-p INDEXER_PLUGINS_BRANCH\t[Optional] wazuh-indexer-plugins repo branch, default is 'master'."
+  echo -e "-r INDEXER_REPORTING_BRANCH\t[Optional] wazuh-indexer-reporting repo branch, default is 'master'."
+  echo -e "-R REVISION\t[Optional] Package revision, default is '0'."
+  echo -e "-s STAGE\t[Optional] Staging build, default is 'false'."
+  echo -e "-d DISTRIBUTION\t[Optional] Distribution, default is 'rpm'."
+  echo -e "-a ARCHITECTURE\t[Optional] Architecture, default is 'x64'."
+  echo -e "-D\tDestroy the docker environment"
+  echo -e "-h\tPrint help"
+}
+
+# ====
+# Main function
+# ====
+function main() {
+    check_project_root_folder $0
+    compose_file="docker/${current}/compose.yml"
+    compose_cmd="docker compose -f $compose_file"
+    REPO_PATH=$(pwd)
+    VERSION=$(cat VERSION)
+    export REPO_PATH
+    export VERSION
+
+    parse_args "${@}"
+    
+    if [[ "$DESTROY" == true || "$DESTROY" == "1" ]]
+    then
+        $compose_cmd down -v
+        exit 0
+    fi
+
+    $compose_cmd up --detach --build
```

Comment:
I think it would be better if we don't _detach_ the container, so we can follow the execution when running the tool, and its easier to use it on a GHA workflow
```suggestion
    $compose_cmd up
```
> The `--build` could also be removed so it only builds when there are changes on the container

<details><summary>Current execution log</summary>

```
bash builder.sh 
[+] Building 1.8s (17/17) FINISHED                                                                                                                docker:desktop-linux
 => [wazuh-indexer-builder internal] load build definition from Dockerfile                                                                                        0.0s
 => => transferring dockerfile: 2.61kB                                                                                                                            0.0s
 => [wazuh-indexer-builder internal] load metadata for docker.io/library/ubuntu:noble                                                                             1.5s
 => [wazuh-indexer-builder internal] load .dockerignore                                                                                                           0.0s
 => => transferring context: 2B                                                                                                                                   0.0s
 => [wazuh-indexer-builder  1/11] FROM docker.io/library/ubuntu:noble@sha256:80dd3c3b9c6cecb9f1667e9290b3bc61b78c2678c02cbdae5f0fea92cc6734ab                     0.0s
 => => resolve docker.io/library/ubuntu:noble@sha256:80dd3c3b9c6cecb9f1667e9290b3bc61b78c2678c02cbdae5f0fea92cc6734ab                                             0.0s
 => [wazuh-indexer-builder internal] load build context                                                                                                           0.0s
 => => transferring context: 5.28kB                                                                                                                               0.0s
 => CACHED [wazuh-indexer-builder  2/11] RUN userdel -r ubuntu                                                                                                    0.0s
 => CACHED [wazuh-indexer-builder  3/11] RUN apt-get update     && apt-get upgrade -y     && apt-get install -y         wget         curl         unzip           0.0s
 => CACHED [wazuh-indexer-builder  4/11] RUN wget -O- https://download.java.net/openjdk/jdk21/ri/openjdk-21+35_linux-x64_bin.tar.gz | tar xz -C /opt/             0.0s
 => CACHED [wazuh-indexer-builder  5/11] RUN wget https://services.gradle.org/distributions/gradle-8.10-bin.zip -P /tmp     && unzip -d /opt/gradle /tmp/gradle-  0.0s
 => CACHED [wazuh-indexer-builder  6/11] RUN apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*                                                       0.0s
 => CACHED [wazuh-indexer-builder  7/11] RUN addgroup --gid 1000 wazuh-indexer && adduser --ingroup wazuh-indexer --uid 1000 wazuh-indexer                        0.0s
 => CACHED [wazuh-indexer-builder  8/11] RUN mkdir -p /repositories/wazuh-indexer-plugins &&   mkdir -p /repositories/wazuh-indexer-reporting &&   chown wazuh-i  0.0s
 => [wazuh-indexer-builder  9/11] COPY docker/builder/entrypoint.sh /                                                                                             0.0s
 => [wazuh-indexer-builder 10/11] RUN chmod +x /entrypoint.sh                                                                                                     0.1s
 => [wazuh-indexer-builder 11/11] WORKDIR /home/wazuh-indexer                                                                                                     0.0s
 => [wazuh-indexer-builder] exporting to image                                                                                                                    0.0s
 => => exporting layers                                                                                                                                           0.0s
 => => exporting manifest sha256:d5274b5981724398ca6e8ce6a70d4e28dcba6be6189153c99ef89f5fa38cffc6                                                                 0.0s
 => => exporting config sha256:972f88fdf27c331269b6c4d5a26102261d76961faf7bc00ebf924753a7c7a5aa                                                                   0.0s
 => => exporting attestation manifest sha256:9a82ca393e389e53bf621f1f47098c4d6b3be389baef14a4dbf107d95d7cacc6                                                     0.0s
 => => exporting manifest list sha256:aba23ec184e479bcd4796aef8af7bca1a7f3e34e52e9c18e332823bd7ce217d3                                                            0.0s
 => => naming to docker.io/library/wazuh-indexer-builder:5.0.0                                                                                                    0.0s
 => [wazuh-indexer-builder] resolving provenance for metadata file                                                                                                0.0s
[+] Running 1/1
 ✔ Container wazuh-indexer-builder_5.0.0  Started                                                                                                                 0.4s 
```

</details>

Reply:
Good points. I'm making those changes.